### PR TITLE
Issue/480 missing paragraph

### LINKED ISSUE #480: Paragraph tag is getting removed
The `<p>` tag is  being removed during parsing. It should be preserved as it may contain important attributes and Aztec should not mess with the HTML in general.

This keeps happening only when Calypso mode is enabled.

### PR DESCRIPTION
**Pull request by 0nko**
Fixes #480.

### CODE REVIEW

**Comment by khaykov** [comment]
I found an issue with paragraph extending on a new line.

Load editor with something like this `<p class="super">paragraph</p>`
Press enter at the end of the line.
Type "test".
Switch to HTML mode and notice that paragraph tag extended to the new line, like this:
```
<p class="super">paragraph

test</p>
```
Switch back to visual and to HTML again, notice HTML became:
```
<p class="super">paragraph</p>
test
```

**Comment by 0nko** [comment]
Ready for a new pass.

File: aztec/src/main/kotlin/org/wordpress/aztec/source/Format.kt [hunk 1]
```diff
@@ -324,20 +325,36 @@ internal object Format {
                 val spanEnd = text.getSpanEnd(it)
 
                 if (text[spanStart] == '\n' && text.getSpans(spanEnd, spanEnd + 1, IAztecParagraphStyle::class.java)
-                        .filter { it !is ParagraphSpan && text.getSpanStart(it) == spanEnd }.isEmpty()) {
+                        .filter { (it !is ParagraphSpan || !it.attributes.isEmpty()) && text.getSpanStart(it) == spanEnd }.isEmpty()) {
```
Comment (maxme) [inline aztec/src/main/kotlin/org/wordpress/aztec/source/Format.kt]: Just to double check, this only retains the `<p>` if it contains at least one attribute, right?

Comment (0nko) [reply aztec/src/main/kotlin/org/wordpress/aztec/source/Format.kt]: That's right.

Comment (maxme) [reply aztec/src/main/kotlin/org/wordpress/aztec/source/Format.kt]: Ok, I updated `testRetainParagraphFormatting` in 90b4b7e

**Comment by maxme** [comment]
Looks good, :shipit:

**Pull request merged**